consensus/istanbul/core/core.go

```diff
@@ -197,6 +197,17 @@ func (c *core) startNewRound(round *big.Int) {
 		logger = c.logger.New("old_round", c.current.Round(), "old_seq", c.current.Sequence())
 	}
 
+	logger.Trace("Start new ibft round")
+
+	// If new round is 0, then check if qibftConsensus needs to be enabled
+	if round.Uint64() == 0 && c.backend.IsQIBFTConsensus() {
+		logger.Trace("Starting qibft consensus as qibftBlock has passed")
+		if err := c.backend.StartQIBFTConsensus(); err != nil {
+			// TODO Do we panic or continue and try to start qibft when processing the next sequence
```

Comment:
what can make QIBFT fail to start? will this create inconsistencies between nodes if we don't panic?

Reply:
This is legacy IBFT to QBFT transition flow, as part of starting QIBFT, we try to stop legacy IBFT and that can return an error. What we decided was to continue if that happens, so that QIBFT can be started for the next sequence. 

I will take out the TODO from there and add another comment explaining this

Reply:
Done